BaseClasses.py

```diff
@@ -516,6 +516,49 @@ def get_spheres(self) -> Iterator[Set[Location]]:
                 state.collect(location.item, True, location)
             locations -= sphere
 
+    def get_sendable_spheres(self) -> Iterator[Set[Location]]:
+        """
+        yields a set of multiserver sendable locations (location.item.code: int) for each logical sphere
+
+        If there are unreachable locations, the last sphere of reachable locations is followed by an empty set,
+        and then a set of all of the unreachable locations.
+        """
+        state = CollectionState(self)
+        locations = set()
+        events = set()
```

Comment:
```suggestion
        locations: Set[Location] = set()
        events: Set[Location] = set()
```

Reply:
PyCharm, at least, correctly inferred all these types inside the method.

Reply:
That doesn't seem like a safe inference.

```python
a = set()
a.add(1)  # inferred set[int] ?
a.add("1")  # inferred set[int | str] ?
a.add(None)  # inferred set[int | str | None] ?
```
Is it not going to stop me from putting anything in this set?
I would like it to make sure I don't put the wrong things into the set.